res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -1984,138 +2119,245 @@ NumarkMixtrack3.PitchBendPlusButton = function(channel, control, value, status,
+                if (value === 127) {
+                    gainIncrement = - gainIncrement;
+                }   
+                                
+                gainValue[i-1] = gainValue[i-1] + gainIncrement;
 
-                    if ((gainValue[j-1] + gainIncrement) <0) {
-                        gainValue[j-1] = 0;
-                    }
+                if ((gainValue[i-1] + gainIncrement) <0) {
+                    gainValue[i-1] = 0;
+                }
+                
+                if ((gainValue[i-1] + gainIncrement) >4) {
+                    gainValue[i-1] = 4;
+                }
                     
-                    if ((gainValue[j-1] + gainIncrement) >4) {
-                        gainValue[j-1] = 4;
-                    }           
-                }   
             }
-
-            // we adjust pregain with adjusted value
-            if (decknum === 1) {
-                for (i = 1; i <= 4; i++) {  
-                    if (gainValue[i-1] >=0 && gainValue[i-1] <= 4) {  
-                        engine.setValue("[Sampler" + i + "]", "pregain", gainValue[i-1]);   
-                    }
+            
+        } else {
+            for (j = 5; j <= 8; j++) {
+                gainValue[j-1] = engine.getValue("[Sampler" + [j] + "]", "pregain");    
+            
+                if (gainValue[j-1] <= 1) { 
+                    // increment value between 0 and 1
+                    gainIncrement = 1/20;  // 20 increments in one full knob turn   
+                }else{
+                    // increment value between 1 and 4
+                    gainIncrement = 3/20;  // 20 increments in one full knob turn
                 }
-            } else {
-                for (i = 5; i <= 8; i++) {
-                    if (gainValue[i-1] >= 0 && gainValue[i-1] <= 4) { 
-                        engine.setValue("[Sampler" + i + "]", "pregain", gainValue[i-1]);   
-                    }
+                
+                // beat knobs sends 1 or 127 as value. If value = 127, turn is counterclockwise, we reduce gain
+                if (value === 127) {
+                    gainIncrement = - gainIncrement;
                 }   
+                
+                gainValue[j-1] = gainValue[j-1] + gainIncrement;
+
+                if ((gainValue[j-1] + gainIncrement) <0) {
+                    gainValue[j-1] = 0;
+                }
+                
+                if ((gainValue[j-1] + gainIncrement) >4) {
+                    gainValue[j-1] = 4;
+                }           
+            }   
+        }
+
+        // we adjust pregain with adjusted value
+        if (decknum === 1) {
+            for (i = 1; i <= 4; i++) {  
+                if (gainValue[i-1] >=0 && gainValue[i-1] <= 4) {  
+                    engine.setValue("[Sampler" + i + "]", "pregain", gainValue[i-1]);   
+                }
             }
+        } else {
+            for (i = 5; i <= 8; i++) {
```

Comment:
Not sure I get it the magic of what you are teaching me... The if statement is related to which deck is the knob used from. What this is doing is if it is deck 1, then adjust gain of samplers 1-4, if deck 2, adjust gain of samplers 5-8. 
If I read properly, the variable "i" will always have the same value, so only the gain of samplers 1-4 would be affected, or all 8 samplers...